its/src/test/java/com/sonar/it/scanner/msbuild/utils/TestUtils.java

```diff
@@ -389,6 +390,7 @@ public static List<Issue> projectIssues(Orchestrator orchestrator, String projec
       results.addAll(issues.getIssuesList());
       page++;
     } while (results.size() < issues.getPaging().getTotal());
+    results.removeIf(x -> !StringUtils.equalsAny(projectKey, x.getProject(), x.getComponent()));
```

Comment:
The component key filtering in the `issues.SearchRequest` does not work on some older SQ versions. Older versions return the full list of issues. We therefore need to do the filtering again on the request result.

Reply:
Can you do this selectively for an old SQ version? So we'd have a piece of code where we'd see when we can remove this nonsense?